corehq/apps/hqadmin/views.py

```diff
@@ -511,10 +511,27 @@ def get_context_data(self, **kwargs):
             string_payload = ''.join(response.streaming_content)
             xml_payload = etree.fromstring(string_payload)
             restore_id_element = xml_payload.find('{{{0}}}Sync/{{{0}}}restore_id'.format(SYNC_XMLNS))
-            num_cases = len(xml_payload.findall('{http://commcarehq.org/case/transaction/v2}case'))
-            num_locations = len(
-                xml_payload.findall("{{{0}}}fixture[@id='locations']/{{{0}}}locations/{{{0}}}location"
-                                    .format(RESPONSE_XMLNS)))
+            cases = xml_payload.findall('{http://commcarehq.org/case/transaction/v2}case')
+            num_cases = len(cases)
+            case_type_counts = dict(Counter(
+                case.find(
+                    '{http://commcarehq.org/case/transaction/v2}create/'
+                    '{http://commcarehq.org/case/transaction/v2}case_type'
+                ).text for case in cases
+            ))
+            locations = xml_payload.findall(
+                "{{{0}}}fixture[@id='locations']/{{{0}}}locations/{{{0}}}location".format(RESPONSE_XMLNS)
+            )
+            num_locations = len(locations)
+            location_type_counts = dict(Counter(location.attrib['type'] for location in locations))
+            reports = xml_payload.findall(
+                "{{{0}}}fixture[@id='commcare:reports']/{{{0}}}reports/".format(RESPONSE_XMLNS)
+            )
+            num_reports = len(reports)
+            report_row_counts = {
+                report.attrib['report_id']: len(report.findall('{{{0}}}rows/{{{0}}}row'.format(RESPONSE_XMLNS)))
+                for report in reports
+            }
```

Comment:
This method is getting pretty big, might be time to pull out these lookups into a helper